botocore/awsrequest.py

```diff
@@ -414,6 +415,25 @@ def reset_stream(self):
             logger.debug("Unable to rewind stream: %s", e)
             raise UnseekableStreamError(stream_object=self.body)
 
+    def prepare_body(self, data, files, json=None):
+        """Prepares the given HTTP body data."""
+        super(AWSPreparedRequest, self).prepare_body(data, files, json)
+
+        # Transfer-Encoding is not supported for AWS Services so remove it
+        # if it is added.
+        if 'Transfer-Encoding' in self.headers:
+            self.headers.pop('Transfer-Encoding')
+
+        # Calculate the Content-Length by trying to seek the file as
+        # requests cannot determine content length for some seekable file-like
+        # objects.
+        if 'Content-Length' not in self.headers:
+            if hasattr(data, 'seek') and hasattr(data, 'tell'):
```

Comment:
Shouldn't we have an error if the data isn't seekable/tellable?  Otherwise we popped the transfer encoding header and didn't replace it with anything.